pkg/extflag/pathorcontent.go

```diff
@@ -44,30 +44,47 @@ func RegisterPathOrContent(cmd FlagClause, flagName string, help string, require
 	}
 }
 
-// Content returns content of the file. Flag that specifies path has priority.
+// Content returns the content of the file when given or directly the content that has passed to the flag.
+// Flag that specifies path has priority.
 // It returns error if the content is empty and required flag is set to true.
 func (p *PathOrContent) Content() ([]byte, error) {
-	contentFlagName := p.flagName
 	fileFlagName := fmt.Sprintf("%s-file", p.flagName)
 
 	if len(*p.path) > 0 && len(*p.content) > 0 {
-		return nil, errors.Errorf("both %s and %s flags set.", fileFlagName, contentFlagName)
+		return nil, errors.Errorf("both %s and %s flags set.", fileFlagName, p.flagName)
 	}
 
 	var content []byte
 	if len(*p.path) > 0 {
 		c, err := ioutil.ReadFile(*p.path)
 		if err != nil {
-			return nil, errors.Wrapf(err, "loading YAML file %s for %s", *p.path, fileFlagName)
+			return nil, errors.Wrapf(err, "loading file %s for %s", *p.path, fileFlagName)
 		}
 		content = c
 	} else {
 		content = []byte(*p.content)
 	}
 
 	if len(content) == 0 && p.required {
-		return nil, errors.Errorf("flag %s or %s is required for running this command and content cannot be empty.", fileFlagName, contentFlagName)
+		return nil, errors.Errorf("flag %s or %s is required for running this command and content cannot be empty.", fileFlagName, p.flagName)
 	}
 
 	return content, nil
 }
+
+// Path returns the path of the given file that has passed to the flag.
+// It returns error if both a path and content are given.
+// It returns error if the required flag is set to true and path is empty.
+func (p *PathOrContent) Path() (string, error) {
```

Comment:
Hmmmm...

So `Content()` is a bit different semantically. We aim for Content to be always there unless flag is optional.

With Path.. we have one more case, of user not specifying path at al, but specifying content. While technically this method should be understandable there are so many cases to tests to check which of those 6 cases we have (content passed, path passed, filag not used) x2 (required or not)

While all of this we have also this important message that is hidden in terms of hashring file:

IF you specify via content it will be used. If specify via path it will be dynamically reloadable. There are 2 problems with this:

1. This is surprising. We never reload content. How user can know we do it now? And let's say we are explicit we reload hashring. How user will know that other flags then hashrings are not dynamically reloadable?
2. What if I want via path but don't want to dynamically reload? For example I want to put file, start receive and remove (for some reason). Maybe rare, but someone could be surprised this is not possible. Might other usecases as well.

I think we need to think this through. I have some ideas, but before I state them,  maybe you have as well? 🤗 Do you see the problem?

Reply:
> This is surprising. We never reload content. How user can know we do it now? And let's say we are explicit we reload hashring. How user will know that other flags then hashrings are not dynamically reloadable?

I understand the concern. Maybe overloading the `PathOrContent` for the Receive's use case is not the best way.
Receive has a special way to handle the given path as you already mentioned. It initializes a watcher for the given path.

> What if I want via path but don't want to dynamically reload? For example I want to put file, start receive and remove (for some reason). Maybe rare, but someone could be surprised this is not possible. Might other usecases as well.

This is a problem we already have. This PR shouldn't be aiming to solve this. However, I can add more documentation to clarify this behaviour.

How about rather than overloading the `PathOrContent`, we implement dedicated logic in the receive. Specific to receive? Would this work? WDYT?

Reply:
hm what about, creating a second constructor?

So we can have:

* `something` or `something-file` as one constructor
* `something` and `something-file` and `something-reloadable-file` if you use another constructor

So receive will use latter (: Then we can actually embed file watching in the second constructor, directly in pathOrContent package 🤗 

WDYT?

Reply:
I like the idea. I would love to work on it on a separate PR to introduce this feature in a more generalized form.
How about merging this PR as is and then creating an issue for it? As I told you I can work on it. WDYT?

Reply:
@bwplotka As we talked off-line, do we have a consensus? Can we proceed with this?

Reply:
YES! love it. Thanks!